frontend/ui/widget/footnotewidget.lua

```diff
@@ -89,6 +89,21 @@ body > li { list-style-type: none; }
 
 /* Remove any (possibly multiple) backlinks in Wikipedia EPUBs footnotes */
 .noprint { display: none; }
+
+/* Attempt to display FB2 footnotes as expected (as crengine does, putting
+ * the footnote number on the same line as the first paragraph via its
+ * support of "display: run-in" and a possibly added autoBoxing element) */
+body > section > autoBoxing > *,
```

Comment:
Bit late since it's already merged, but that `body >` part looks a bit unnecessary?

Reply:
Well, not necessary to catch these FB2 footnotes.
But having them, as we know these FB2 footnotes are pure SECTION > (TITLE/P), and they'll be child of the HTMLBoxWidget added BODY, having them like this, we'll be sure to NOT have that applied on any (probably super rare) DIV > SECTION > TITLE/P that we could find in the inner content of some EPUB footnotes :)

Reply:
If you can end up with div > section couldn't you also end up with body > section?

Reply:
Yes, we could...
But in this case, I can't, here, distinguish if it comes from a FB2 footnote (and need these CSS tweaks) or if it comes from some EPUB footnote (and does need it).
But in the other case DIV > SECTION, I can.
So, just doing what I can :)